lib/openstax/aws/stack.rb

```diff
@@ -338,6 +338,52 @@ def status
       end
     end
 
+    def status_reason
+      begin
+        aws_stack.stack_status_reason
+      rescue Aws::CloudFormation::Errors::ValidationError => ee
+        case ee.message
+        when /Stack.*does not exist/
+          "DOES_NOT_EXIST"
+        else
+          raise
+        end
+      end
+    end
+
+    def events
+      begin 
+        aws_stack.events()
+      rescue
+        raise "something's wrong sorry"
+      end
+    end
+
+    def latest_failed_events
+      latest_deployment_reasons = []
+      begin
+        ###search in latest deployment events
+        aws_stack.events.each do |event|
```

Comment:
You could replace this with `.each_with_object([])` make it little cleaner. It would also remove the need to `return` right after it